h2o-py/h2o/frame.py

```diff
@@ -1011,6 +1011,34 @@ def set_levels(self, levels):
         return H2OFrame._expr(expr=ExprNode("setDomain", self, False, levels), cache=self._ex._cache)
 
 
+    def rename(self, columns=None):
+        """
+        Change names of columns in the frame.
+
+        Dict key is an index or name of the column whose name is to be set.
+        Dict value is the new name of the column.
+
+        :param columns: dict-like transformations to apply to the column names
+        """
+        assert_is_type(columns, None, dict)
+        new_names = self.names
+        ncols = self.ncols
+
+        for col, name in columns.items():
+            col_index = None
+            if is_type(col, int) and (-ncols <= col < ncols):
+                col_index = (col + ncols) % ncols  # handle negative indices
+            elif is_type(col, str) and col in self.names:
+                col_index = self.names.index(col)  # lookup the name
+
+            if col_index is not None:
```

Comment:
Should we show an error if the column is not found?

Reply:
We can do that and we are doing that with `set_names`. Pandas `rename` behavior is to ignore any columns that are not found. We can report errors to be consistent with `set_names`, up to you guys.

Reply:
Consistency with Pandas is preferred!